aspnet/fundamentals/caching/sample/test/CachingSample.Tests/MemoryCacheTests.cs

```diff
@@ -0,0 +1,93 @@
+﻿using System.Threading;
+using Microsoft.Extensions.Caching.Memory;
+using Microsoft.Extensions.Primitives;
+using Xunit;
+
+namespace CachingSample.Tests
+{
+    public class MemoryCacheTests
+    {
+        private readonly MemoryCache _memoryCache;
+        private readonly string _cacheItem = "value";
+        private readonly string _cacheKey = "key";
+        private string _result;
+
+        public MemoryCacheTests()
+        {
+            _memoryCache = new MemoryCache(new MemoryCacheOptions());
+        }
+
+        [Fact]
+        public void RemovalFiresCallback()
+        {
+            _memoryCache.Set(_cacheKey, _cacheItem,
+                new MemoryCacheEntryOptions()
+                .RegisterPostEvictionCallback(
+                    (key, value, reason, substate) =>
+                    {
+                        _result = $"'{key}':'{value}' was evicted because: {reason}";
+                    }
+                ));
+
+            _memoryCache.Remove(_cacheKey);
+
+            // wait for it
+            Thread.Sleep(100);
+
+            Assert.Equal("'key':'value' was evicted because: Removed", _result);
+        }
+
+        [Fact]
+        public void CancellationTokenFiresCallback()
+        {
+            var cts = new CancellationTokenSource();
+            _memoryCache.Set(_cacheKey, _cacheItem,
+                new MemoryCacheEntryOptions()
+                .AddExpirationToken(new CancellationChangeToken(cts.Token))
+                .RegisterPostEvictionCallback(
+                    (key, value, reason, substate) =>
+                    {
+                        _result = $"'{key}':'{value}' was evicted because: {reason}";
+                    }
+                ));
+
+            // trigger the token
+            cts.Cancel();
+
+            // wait for it
+            Thread.Sleep(100);
```

Comment:
Replaced with ManualResetEvent; tests pass; let me know if there's a better way for me to implement MRE than what I did.